httpx/models.py

```diff
@@ -482,17 +486,31 @@ def __eq__(self, other: typing.Any) -> bool:
             return False
         return sorted(self._list) == sorted(other._list)
 
+    def copy(self: "Headers", hide_sensitive_items: bool = False) -> "Headers":
+        clone = copy.deepcopy(self)
+        if hide_sensitive_items:
+            for hidden_key in ("authorization", "proxy-authorization", "password"):
+                if hidden_key in clone:
+                    prefix = (
+                        clone[hidden_key].split(" ")[0]
+                        if len(clone[hidden_key].split(" ")) > 0
+                        and "authorization" in hidden_key
+                        else ""
+                    )
+                    clone[hidden_key] = f"{prefix} ***"
+        return clone
+
     def __repr__(self) -> str:
         class_name = self.__class__.__name__
 
         encoding_str = ""
         if self.encoding != "ascii":
             encoding_str = f", encoding={self.encoding!r}"
-
-        as_dict = dict(self.items())
-        if len(as_dict) == len(self):
+        clone = self.copy(hide_sensitive_items=True)
+        as_dict = dict(clone.items())
```

Comment:
What do you think about grabbing the items and filtering their value based on being included in a list:

ie:
```python
sensitive_headers = {"authorization", "proxy-authorization"}
clone = [(k, "[secure]" if k in sensitive_headers else v for k, v in self.items()]
```

Would this work the same?